extensions/flight-sql/build.gradle

```diff
@@ -0,0 +1,71 @@
+plugins {
+    id 'java-library'
+    id 'io.deephaven.project.register'
+}
+
+description = 'The Deephaven Flight SQL library'
+
+sourceSets {
+    jdbcTest {
+        compileClasspath += sourceSets.main.output
+        compileClasspath += sourceSets.test.output
+
+        runtimeClasspath += sourceSets.main.output
+        runtimeClasspath += sourceSets.test.output
```

Comment:
None of this seems to be necessary - both in that jdbcTest doesnt depend on test at all (should be using the `java-test-fixtures` plugin to share like that), and that the remaining lines can be a oneliner dependency below:
```
    jdbcTestImplementation project(':extensions-flight-sql')
```

Reply:
I don't think `java-test-fixture` is necessarily the correct model... that said, I've divorced the jdbcTest source set from the test source set. There may be better ways to do this... but I _think_ what I have is functionally correct. Also added some comments about the differences between test and jdbcTest.